bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/parallacticangle/ParallacticAngleControls.scala

```diff
@@ -142,32 +143,42 @@ class ParallacticAngleControls(isPaUi: Boolean) extends GridBagPanel with Publis
         * This should be called whenever the position angle or parallactic angle changes.
         * A warning icon and tooltip are displayed if the two are different.
         */
-      def warningState(): Unit = Swing.onEDT {
+      def warningState(): Unit =
         for {
           e     <- editor
           fmt   <- formatter
           inst  <- Option(e.getContextInstrumentDataObject).collect { case s: SPInstObsComp => s }
         } warningStateFromPAString(inst.getPosAngleDegrees.toString)
-      }
 
-      def warningStateFromPAString(paStr: String): Unit = Swing.onEDT {
+      def warningStateFromPAString(paStr: String): Unit =
         for {
           e     <- editor
           fmt   <- formatter
           inst  <- Option(e.getContextInstrumentDataObject).collect { case s: SPInstObsComp => s }
-          angle <- parallacticAngle
         } {
-          val explicitlySet = !fmt.format(angle.toDegrees).equals(paStr) &&
-                              !fmt.format((angle + Angle.fromDegrees(180)).toDegrees).equals(paStr)
-          if (explicitlySet) {
+          // We warn only if the parallactic angle exists and has been explicitly set.
+          val warningFlag = parallacticAngle.exists { angle =>
+              val fa = fmt.format(angle.toDegrees)
+              val faFlip = fmt.format(angle.flip.toDegrees)
+              !fa.equals(paStr) && !faFlip.equals(paStr)
+          }
+
+          if (warningFlag) {
             icon = Resources.getIcon("eclipse/alert.gif")
```

Comment:
These calls should definitely be done on the EDT. How do you know this is called from inside swing